cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/Scheduler.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.scheduler;
+
+import co.cask.cdap.common.AlreadyExistsException;
+import co.cask.cdap.common.NotFoundException;
+import co.cask.cdap.internal.app.runtime.schedule.ProgramSchedule;
+import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.proto.id.ProgramId;
+import co.cask.cdap.proto.id.ScheduleId;
+
+import java.util.List;
+
+/**
+ * Scheduler interface for deploying and retrieving schedules.
+ */
+public interface Scheduler {
+
+  /**
+   * Add a schedule to the store.
+   *
+   * @param schedule the schedule to add
+   * @throws AlreadyExistsException if the schedule already exists
+   */
+  void addSchedule(ProgramSchedule schedule) throws AlreadyExistsException;
+
+  /**
+   * Add one or more schedules to the store.
+   *
+   * @param schedules the schedules to add
+   * @throws AlreadyExistsException if one of the schedules already exists
+   */
+  void addSchedules(Iterable<? extends ProgramSchedule> schedules) throws AlreadyExistsException;
+
+  /**
+   * Removes a schedule from the store. Succeeds whether the schedule exists or not.
+   *
+   * @param scheduleId the schedule to delete
+   */
+  void deleteSchedule(ScheduleId scheduleId);
```

Comment:
ok I changed that. One thing to remember is that deleting all schedules for an app will not throw NotFound even if the app does not exist. Because the schedule store does not know that, it only knows whether the app has schedules.